gradle/setup.gradle

```diff
@@ -106,10 +134,41 @@ publishing {
 			}
 		}
 	}
+	if (qualifyVersion("$version") == "RELEASE") {
+		repositories {
+			maven {
+				name = "sonatype"
+				url = "https://s01.oss.sonatype.org/service/local/staging/deploy/maven2"
+				credentials {
+					username findProperty("sonatypeUsername")
+					password findProperty("sonatypePassword")
+				}
+			}
+		}
+	}
+}
+
+if (rootProject.hasProperty("artifactory_publish_password")) {
+	apply plugin: "com.jfrog.artifactory"
+
+	artifactoryPublish {
+		publications(publishing.publications.mavenJava)
+	}
 }
 
-artifactoryPublish {
-	//don't activate the onlyIf clause below yet, so that Bamboo builds will publish stuff...
-	//onlyIf { project.hasProperty("artifactory_publish_password")") }
-	publications(publishing.publications.mavenJava)
+if (qualifyVersion("$version") in ["RELEASE", "MILESTONE"]) {
+	apply plugin: 'signing'
+
+	signing {
+		//requiring signature if there is a publish task that is not to MavenLocal
+		required {  gradle.taskGraph.allTasks.any { it.name.toLowerCase().contains("publish")	&& !it.name.contains("MavenLocal") } }
```

Comment:
IIRC when applying the same logic to BlockHound, it was too early to have access to the taskGraph

Reply:
at best I can see, this looks like it works. will remove in case of problems last minute when performing the actual releases